test/e2e/node_feature_discovery_test.go

```diff
@@ -221,15 +222,27 @@ var _ = SIGDescribe("NFD master and worker", func() {
 
 				// Launch nfd-master
 				By("Creating nfd master pod and nfd-master service")
-				podSpecOpts := createPodSpecOpts(
-					testpod.SpecWithContainerImage(dockerImage()),
-					testpod.SpecWithTolerations(testTolerations),
-					testpod.SpecWithContainerExtraArgs("-enable-taints"),
-					testpod.SpecWithContainerExtraArgs(
-						"-deny-label-ns=*.denied.ns,random.unwanted.ns,*.vendor.io",
-						"-extra-label-ns=custom.vendor.io",
-					),
-				)
+				var podSpecOpts []testpod.SpecOption
+				if customMasterPodSpecOpts == nil {
+					podSpecOpts = createPodSpecOpts(
+						testpod.SpecWithContainerImage(dockerImage()),
+						testpod.SpecWithTolerations(testTolerations),
+						testpod.SpecWithContainerExtraArgs("-enable-taints"),
+						testpod.SpecWithContainerExtraArgs(
+							"-deny-label-ns=*.denied.ns,random.unwanted.ns,*.vendor.io",
+							"-extra-label-ns=custom.vendor.io",
+						),
+					)
+				} else {
+					cm := testutils.NewConfigMap("nfd-master-conf", "nfd-master.conf", `
+denyLabelNs: ["*.denied.ns","random.unwanted.ns"]
+`)
+					_, err = f.ClientSet.CoreV1().ConfigMaps(f.Namespace.Name).Create(context.TODO(), cm, metav1.CreateOptions{})
+					Expect(err).NotTo(HaveOccurred())
```

Comment:
These can and should be done in the BeforeEach() method of the test that needs the configmap